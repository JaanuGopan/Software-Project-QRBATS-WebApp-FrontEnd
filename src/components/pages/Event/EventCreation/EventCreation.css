.event-main-container1 {
    width: 60%;
    border-top: 6px solid #0077C0;
    border-radius: 10px;
    padding: 20px;
    background-color: #ffffff;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    z-index: 3;
}

.event-main-container2 {
    margin-top: 10px;
    width: 100%;
    border-top: 6px solid #0077C0;
    border-radius: 10px;
    padding: 20px;
    background-color: #ffffff;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    z-index: 3;
}

.eventCreation-field{
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;
    flex-wrap: wrap;
}

.Create-logo{
    width: 50%;
    min-width: 150px;
    max-width: 300px;
    border-radius: 10px;
}

.eventCreation-input-field{
    padding: 5%;
}

h2 {
    text-align: center;
}

.form-container {
    flex: 1;
}

.form-group {
    margin-bottom: 15px;
    position: relative;
}

label {
    display: block;
    margin-bottom: 5px;
}
/*
input[type="text"],
input[type="password"] {
    width: 100%;
    padding: 10px;
    border: none;
    border-bottom: 3px solid #0077C0;
    border-radius: 5px;
    background-color: #ffffff;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
}

.input-field {
    padding-left: 40px;
}

.input-with-icon {
    display: flex;
    align-items: center;
}

.input-with-icon .input-icon {
    margin-right: 10px;
}

.input-with-icon .react-time-picker {
    width: 100px;
}*/

input[type="text"]:focus,
input[type="password"]:focus {
    outline: none;
    border-color: #007bff;
}

input[type="text"]:not(:focus),
input[type="password"]:not(:focus) {
    border-radius: 5px 5px 0 0;
}

input[type="text"]::placeholder,
input[type="password"]::placeholder {
    color: #aaa;
}

.back-button {
    background: #0077C0;
    border: none;
    padding: 10px;
    cursor: pointer;
    outline: none;
    margin-right: 10px;
    border-radius: 50%;
    transition: background-color 0.3s ease;
    display: flex;
    align-items: center;
    justify-content: center;
}

.back-button:hover {
    background: rgba(92, 92, 165, 0.8);
}

.back-button .icon {
    color: #fff;
}

.event-login-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    margin-top: 20px;
    position: relative;
}

.button-group {
    margin-bottom: 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}

.eventCreation-switchbutton {
    margin-right: 30px;
}

button {
    padding: 10px 20px;
    background-color: #0077C0;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
}

a {
    color: #0077C0;
    text-decoration: none;
}

button:hover,
a:hover {
    background-color: #0056b3;
}

.date-label text{
    font-size: small;
}

.qr-code{
    margin-top: 10vh;
    flex: 0 0 85% 0;
    height: 100%;
    width: 100%;
    padding: 10px;
    max-height: 80vh;
    align-items: center;
    justify-content: center;
}

.row-center{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 10px;
}

.QRbutton{
    margin-top: 10px;
    margin-right: 10px;
}

.closeCreateEventWindow{
  color: red;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: flex-end;
  padding: 5px;
  border-radius: 50px;
  margin-left: 95%;
  width: 25px;
}
.closeCreateEventWindow:hover{
    color: white;
  background-color: red;
}

.QRCode-img{
    width: 100%;
    height: 100%;
    
}