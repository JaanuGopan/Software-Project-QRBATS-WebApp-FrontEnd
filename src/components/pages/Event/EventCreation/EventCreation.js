import "./EventCreation.css";
import eventCreationImage from "../../../../assets/Images/signin/Signin.jpeg";
import { FaUser, FaCalendarAlt, FaMapMarkerAlt, FaTimes } from "react-icons/fa";
import { useNavigate } from "react-router-dom";

import React, { useState, useRef } from "react";
import { useLocation } from "react-router-dom";
import QRCode from "react-qr-code";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";

function EventCreation() {
  const navigate = useNavigate();

  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState("");
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState("Event");

  const qrCodeRef = useRef(null);

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (
      eventDate === "" ||
      eventName === "" ||
      eventTime === "" ||
      eventVenue === ""
    ) {
      alert("Please fill all the fields");
      return;
    } else {
      try {
        const response = await axios.post(
          "http://localhost:8080/api/v1/auth/create",
          {
            eventName: eventName,
            eventDate: eventDate,
            eventValidDate: eventValidDate,
            eventTime: eventTime,
            eventVenue: eventVenue,
            eventRole: "LECTURE",
            eventAssignedStaffId: 1,
            eventModuleName: moduleName,
            eventAssignedUserId: 1,
          }
        );
        const responseEventName = response.data.eventName;
        console.log("Event is : " + responseEventName);
      } catch (error) {
        console.error("Event failed", error);
      } finally {
        setShowQRCode(true);
      }
    }
  };

  const handleShareQRCode = () => {
    html2canvas(qrCodeRef.current).then((canvas) => {
      const imgData = canvas.toDataURL("image/png");
      const pdf = new jsPDF();
      pdf.addImage(imgData, "PNG", 0, 0);
      pdf.save("qr_code.pdf");
    });
  };

  // Concatenate all event details into a single string
  const eventDetails = `${eventName}, ${moduleName}, ${eventDate}, ${eventTime}, ${eventVenue}`;

  return (
    <div className="event-main-div">
      <div className="event-main-container1">
        <h2>Create {title}</h2>
        <div className="eventCreation-field">
          <div className="image-container">
            <img src={eventCreationImage} className="logo" alt="Logo" />
          </div>
          <div className="eventCreation-input-field">
            <div className="button-group">
              <button
                className="eventCreation-switchbutton"
                onClick={() => setShowModuleNameInput(false)}
              >
                Event
              </button>
              <button
                className="eventCreation-switchbutton"
                onClick={() =>
                  setShowModuleNameInput(true) && setTitle("Lecture")
                }
              >
                Lecture
              </button>
            </div>
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <div className="input-with-icon">
                  <input
                    type="text"
                    id="eventName"
                    name="eventName"
                    placeholder={title + " Name"}
                    className="form-control mb-2"
                    value={eventName}
                    onChange={(e) => setEventName(e.target.value)}
                  />
                </div>
              </div>
              {showModuleNameInput && (
                <div className="form-group">
                  <div className="input-with-icon">
                    <input
                      type="text"
                      id="moduleName"
                      name="moduleName"
                      placeholder="Module Name"
                      className="form-control mb-2"
                      value={moduleName}
                      onChange={(e) => setModuleName(e.target.value)}
                    />
                  </div>
                </div>
              )}
              <div className="date-div">
                <div className="form-group">
                  <div>
                    <label className="date-label" htmlFor="eventDate">
                      {title} Starting Date
                    </label>
                    <input
                      type="date"
                      value={eventDate}
                      onChange={(e) => setEventDate(e.target.value)}
                      className="form-control mb-2"
                    />
                  </div>
                </div>
                <div className="form-group">
                  <div>
                    <label className="date-label" htmlFor="eventDate">
                      {title} Ending Date
                    </label>
                    <input
                      type="date"
                      value={eventValidDate}
                      onChange={(e) => setEventValidDate(e.target.value)}
                      className="form-control mb-2"
                    />
                  </div>
                </div>
              </div>
              <div className="form-group">
                <div>
                  <select
                    value={eventVenue}
                    onChange={(e) => setEventVenue(e.target.value)}
                    className="form-control mb-2"
                  >
                    <option value="">Select Venue</option>
                    <option value="LT1">LT1</option>
                    <option value="LT2">LT2</option>
                    <option value="Auditorium">Auditorium</option>
                    <option value="NCC">NCC</option>
                    <option value="Other">Other</option>
                  </select>
                </div>
              </div>
              <div className="form-group">
                <div>
                  <label className="form-label" htmlFor="eventDate">
                    {title} Starting Time
                  </label>
                  <input
                    type="time"
                    value={eventTime}
                    onChange={(e) => setEventTime(e.target.value)}
                    placeholder="Event Time"
                    className="form-control mb-2"
                  />
                </div>
              </div>

              <div className="form-group"></div>
              <button type="submit" className="btn btn-primary w-100">
                Create {title}
              </button>
            </form>
          </div>
        </div>
        {showQRCode && (
          <div ref={qrCodeRef} className="qr-code">
            <QRCode name="QRCode" value={eventDetails} className="mb-2" />
            <p>Scan this QR code to mark attendance.</p>
            <div className="mt-4">
              <button
                onClick={handleShareQRCode}
                className="btn btn-primary mr-2"
              >
                Share QR Code
              </button>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}

export default EventCreation;
