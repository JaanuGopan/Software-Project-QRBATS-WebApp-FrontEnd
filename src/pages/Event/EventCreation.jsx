import './EventCreation/EventCreation.css';
import eventCreationImage from '../../assets/Images/signin/Signin.jpeg';
import React, { useState, useRef, useContext } from 'react';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import toast, { Toaster } from 'react-hot-toast';
import QRCode from 'qrcode.react';
import InputList from '../../components/textfields/InputList/InputList';
import InputField from '../../components/textfields/InputBox/InputField';
import EventService from '../../api/services/EventService';
import { CircularProgress } from '@mui/material';
import { AuthContext } from '../../config/AuthProvider';

const EventCreationComponent = () => {
  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const [eventId, setEventId] = useState('');
  const [eventName, setEventName] = useState('');
  const [moduleName, setModuleName] = useState(null);
  const [eventDate, setEventDate] = useState('');
  const [eventValidDate, setEventValidDate] = useState('');
  const [eventTime, setEventTime] = useState('');
  const [eventEndTime, setEventEndTime] = useState('');
  const [eventVenue, setEventVenue] = useState('');
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState('EVENT');
  const [eventAssignedUserId, setEventAssignedUserId] = useState(null);

  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState('Event');

  const venueList = ['NCC', 'LT1', 'LT2', 'Auditorium', 'DEIE', 'DMME', 'DCEE', 'Other'];

  const notifySuccess = () => toast.success('Successfully Event Created!');

  const handleInputValidation = () => {
    if (!eventName || !eventDate || !eventValidDate || !eventTime || !eventEndTime || !eventVenue) {
      return false;
    }
    return true;
  };

  const clearEventDetails = () => {
    setEventName('');
    setEventDate('');
    setEventValidDate('');
    setEventTime('');
    setEventEndTime('');
    setEventVenue('');
    setEventAssignedUserId(null);
    setEventRole('EVENT');
  };

  const [processing, setProcessing] = useState(false);

  const handleCreateEvent = async (e) => {
    try {
      setProcessing(true);
      const response = await EventService.saveEvent(
        eventName,
        eventDate,
        eventValidDate,
        eventTime,
        eventEndTime,
        eventVenue,
        eventRole,
        moduleName,
        userId
      );
      setEventId(response.eventId);
      const responseEventName = response.eventName;
      if (response) {
        setShowQRCode(true);
      }
      notifySuccess();
    } catch (error) {
      console.error('Event failed', error);
    } finally {
      setProcessing(false);
    }
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (handleInputValidation()) {
      await handleCreateEvent(e);
    } else {
      console.log('Please fill all the fields');
    }
  };

  const qrCodeRef = useRef(null);
  const downloadQRCode = () => {
    const qrCodeURL = document
      .getElementById('qrCodeEl')
      .toDataURL('image/png')
      .replace('image/png', 'image/octet-stream');
    let aEl = document.createElement('a');
    aEl.href = qrCodeURL;
    aEl.download = 'QR_Code.png';
    document.body.appendChild(aEl);
    aEl.click();
    document.body.removeChild(aEl);
  };

  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  const eventDetails = {
    eventId: eventId,
    eventName: eventName,
    moduleName: moduleName,
    eventDate: eventDate,
    eventValidDate: eventValidDate,
    eventTime: eventTime,
    eventEndTime: eventEndTime,
    eventVenue: eventVenue,
    eventAssignedUserId: userId,
  };
  const qrCodeDetails = JSON.stringify(eventDetails);

  return (
    <div className="event-main-container2">
      <Toaster />
      <h2>Create Event</h2>
      <div className="eventCreation-field">
        <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        <div className="eventCreation-input-field">
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <InputField
                value={eventName}
                placeholder={'Event Name'}
                onChange={(e) => setEventName(e.target.value)}
                inputType={'text'}
              />
            </div>
            <div className="date-div">
              <div className="eventCreation-form">
                <label className="date-label" htmlFor="eventDate">
                  Event Starting Date
                </label>
                <InputField
                  value={eventDate}
                  onChange={(e) => setEventDate(e.target.value)}
                  inputType={'date'}
                />
              </div>
              <div className="eventCreation-form">
                <label className="date-label" htmlFor="eventDate">
                  Event Ending Date
                </label>
                <InputField
                  value={eventValidDate}
                  onChange={(e) => setEventValidDate(e.target.value)}
                  placeholder={'Event Valid Date'}
                  inputType={'date'}
                />
              </div>
            </div>
            <div className="eventCreation-form">
              <label className="date-label" htmlFor="eventDate">
                Venue
              </label>
              <InputList
                list={venueList}
                onChange={(e) => setEventVenue(e.target.value)}
                placeholder={'Venue'}
                value={eventVenue}
                initialValue={'Select Venue'}
              />
            </div>
            <div className="eventCreation-form">
              <label className="form-label" htmlFor="eventDate">
                Event Starting Time
              </label>
              <InputField
                inputType={'time'}
                value={eventTime}
                onChange={(e) => setEventTime(e.target.value)}
                placeholder={'Event Time'}
              />

              <div className="eventCreation-form">
                <label className="form-label" htmlFor="enevtEndTime">
                  Event Ending Time
                </label>
                <InputField
                  inputType={'time'}
                  value={eventEndTime}
                  onChange={(e) => setEventEndTime(e.target.value)}
                  placeholder={'Event End Time'}
                />
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <div className="d-flex justify-content-between mr-3 mt-3">
              {processing ? (
                <CircularProgress />
              ) : (
                <button
                  onClick={() => setQrCodeWindow(true)}
                  type="submit"
                  className="btn btn-primary w-90"
                >
                  Create Event
                </button>
              )}
              <button onClick={clearEventDetails} className="btn btn-danger w-30">
                Clear
              </button>
            </div>
          </form>
        </div>
        {qrCodeWindow && showQRCode && (
          <div className="Admin-Create-Event-Dashboard">
            <div className="event-main-container1">
              <div
                className="closeCreateEventWindow"
                onClick={() => {
                  setQrCodeWindow(false);
                  clearEventDetails();
                }}
              >
                <IoMdCloseCircleOutline />
              </div>
              <h2>Successfully Event Created</h2>
              <div className="row-center">
                <QRCode
                  size={200}
                  id="qrCodeEl"
                  name="QRCode"
                  value={qrCodeDetails}
                  ref={qrCodeRef}
                  className="mb-2"
                />
              </div>
              <div>
                <p className="text-center">Scan this QR code to join {title}</p>
              </div>
              <div className="row-center">
                <div className="QRbutton">
                  <button onClick={downloadQRCode} className="btn btn-success mr-3">
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default EventCreationComponent;
