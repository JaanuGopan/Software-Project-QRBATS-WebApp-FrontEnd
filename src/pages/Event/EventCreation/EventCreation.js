import "./EventCreation.css";
import eventCreationImage from "../../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import QRCode from "react-qr-code";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import toast, { Toaster } from "react-hot-toast";

function EventCreation() {
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState(null);
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("EVENT");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(null);

  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState("Event");

  const qrCodeRef = useRef(null);

  const notifySuccess = () => toast.success("Successfully Event Created!");

  const handleEreseValu = () => {
    setEventName("");
    setModuleName(null);
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventVenue("");
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (
      eventDate === "" ||
      eventName === "" ||
      eventTime === "" ||
      eventVenue === ""
    ) {
      alert("Please fill all the fields");
      return;
    } else {
      try {
        const response = await axios.post(
          "http://localhost:8080/api/v1/event/create",
          {
            eventName: eventName,
            eventDate: eventDate,
            eventValidDate: eventValidDate,
            eventTime: eventTime,
            eventVenue: eventVenue,
            eventRole: eventRole,
            eventModuleName: moduleName,
            eventAssignedUserId: 1,
          }
        );
        const responseEventName = response.data.eventName;
        console.log("Event is : " + responseEventName);
        notifySuccess();
      } catch (error) {
        console.error("Event failed", error);
      } finally {
        setShowQRCode(true);
      }
    }
  };

  const handleDownloadQRCode = () => {
    const canvas = qrCodeRef.current.querySelector("canvas");
    const url = canvas
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    const link = document.createElement("a");
    link.download = "qr_code.png";
    link.href = url;
    link.click();
  };

  const handleShareQRCode = () => {
    html2canvas(qrCodeRef.current).then((canvas) => {
      const imgData = canvas.toDataURL("image/png");
      const pdf = new jsPDF();
      pdf.addImage(imgData, "PNG", 0, 0);
      pdf.save("qr_code.pdf");
    });
  };

  // Concatenate all event details into a single string
  const eventDetails = `${eventName}, ${moduleName}, ${eventDate}, ${eventValidDate}, ${eventTime}, ${eventVenue}, ${eventAssignedUserId}`;

  return (
    <div className="event-main-div">
      <Toaster position="top-center" reverseOrder={false} />
      <div className="event-main-container1">
        <h2>Create {title}</h2>
        <div className="eventCreation-field">
          <div className="image-container">
            <img src={eventCreationImage} className="logo" alt="Logo" />
          </div>
          <div className="eventCreation-input-field">
            <div className="button-group">
              <button
                className="eventCreation-switchbutton"
                onClick={() =>
                  setShowModuleNameInput(false) &&
                  setTitle("Event") &&
                  setEventRole("EVENT") &&
                  handleEreseValu()
                }
              >
                Event
              </button>
              <button
                className="eventCreation-switchbutton"
                onClick={() =>
                  setShowModuleNameInput(true) &&
                  setTitle("Lecture") &&
                  setEventRole("LECTURE") &&
                  handleEreseValu()
                }
              >
                Lecture
              </button>
            </div>
            <form onSubmit={handleSubmit}>
              <div className="eventCreation-form">
                <div className="input-with-icon">
                  <input
                    type="text"
                    id="eventName"
                    name="eventName"
                    placeholder={title + " Name"}
                    className="form-control mb-2"
                    value={eventName}
                    onChange={(e) => setEventName(e.target.value)}
                  />
                </div>
              </div>
              {showModuleNameInput && (
                <div className="eventCreation-form">
                  <div className="input-with-icon">
                    <input
                      type="text"
                      id="moduleName"
                      name="moduleName"
                      placeholder="Module Name"
                      className="form-control mb-2"
                      value={moduleName}
                      onChange={(e) => setModuleName(e.target.value)}
                    />
                  </div>
                </div>
              )}
              <div className="date-div">
                <div className="eventCreation-form">
                  <div>
                    <label className="date-label" htmlFor="eventDate">
                      {title} Starting Date
                    </label>
                    <input
                      type="date"
                      value={eventDate}
                      onChange={(e) => setEventDate(e.target.value)}
                      className="form-control mb-2"
                    />
                  </div>
                </div>
                <div className="eventCreation-form">
                  <div>
                    <label className="date-label" htmlFor="eventDate">
                      {title} Ending Date
                    </label>
                    <input
                      type="date"
                      value={eventValidDate}
                      onChange={(e) => setEventValidDate(e.target.value)}
                      className="form-control mb-2"
                    />
                  </div>
                </div>
              </div>
              <div className="eventCreation-form">
                <div>
                  <select
                    value={eventVenue}
                    onChange={(e) => setEventVenue(e.target.value)}
                    className="form-control mb-2"
                  >
                    <option value="">Select Venue</option>
                    <option value="LT1">LT1</option>
                    <option value="LT2">LT2</option>
                    <option value="Auditorium">Auditorium</option>
                    <option value="NCC">NCC</option>
                    <option value="Other">Other</option>
                  </select>
                </div>
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="eventDate">
                    {title} Starting Time
                  </label>
                  <input
                    type="time"
                    value={eventTime}
                    onChange={(e) => setEventTime(e.target.value)}
                    placeholder="Event Time"
                    className="form-control mb-2"
                  />
                </div>
              </div>

              <div className="eventCreation-form"></div>
              <button type="submit" className="btn btn-primary w-100">
                Create {title}
              </button>
            </form>
          </div>
        </div>
        {showQRCode && (
          <div ref={qrCodeRef} className="qr-code">
            <div className="row-center">
              <QRCode name="QRCode" value={eventDetails} className="mb-2" />
            </div>
            <div>
              <p className="text-center">Scan this QR code to join {title}</p>
            </div>

            <div className="row-center">
              <div className="QRbutton">
                <button
                  onClick={handleShareQRCode}
                  className="btn btn-primary mr-3"
                >
                  Download QR-Code
                </button>
              </div>
              <div className="QRbutton">
                <button
                  onClick={handleShareQRCode}
                  className="btn btn-success mr-3"
                >
                  Save
                </button>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}

export default EventCreation;
