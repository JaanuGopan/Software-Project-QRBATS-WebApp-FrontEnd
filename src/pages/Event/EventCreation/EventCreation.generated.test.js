import renderer from 'react-test-renderer';

import eventCreationImage from "../../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import QRCode from "react-qr-code";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import toast, { Toaster } from "react-hot-toast";
import EventCreation from "./EventCreation";

jest.mock("./EventCreation.css");
jest.mock("../../../assets/Images/signin/Signin.jpeg");
jest.mock("react-qr-code");
jest.mock("axios");
jest.mock("html2canvas");
jest.mock("jspdf");
jest.mock("react-hot-toast");

const renderTree = tree => renderer.create(tree);
describe('<EventCreation>', () => {
  it('should render component', () => {
    expect(renderTree(<EventCreation 
    />).toJSON()).toMatchSnapshot();
  });
  
});