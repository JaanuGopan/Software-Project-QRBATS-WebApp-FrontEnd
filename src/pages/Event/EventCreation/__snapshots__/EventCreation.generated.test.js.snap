// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventCreation> should render component 1`] = `
<div
  className="event-main-div"
>
  <div
    className="event-main-container1"
  >
    <h2>
      Create 
      Event
    </h2>
    <div
      className="eventCreation-field"
    >
      <div
        className="image-container"
      >
        <img
          alt="Logo"
          className="logo"
          src="Signin.jpeg"
        />
      </div>
      <div
        className="eventCreation-input-field"
      >
        <div
          className="button-group"
        >
          <button
            className="eventCreation-switchbutton"
            onClick={[Function]}
          >
            Event
          </button>
          <button
            className="eventCreation-switchbutton"
            onClick={[Function]}
          >
            Lecture
          </button>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="eventCreation-form"
          >
            <div
              className="input-with-icon"
            >
              <input
                className="form-control mb-2"
                id="eventName"
                name="eventName"
                onChange={[Function]}
                placeholder="Event Name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="eventCreation-form"
          >
            <div
              className="input-with-icon"
            >
              <input
                className="form-control mb-2"
                id="moduleName"
                name="moduleName"
                onChange={[Function]}
                placeholder="Module Name"
                type="text"
                value={null}
              />
            </div>
          </div>
          <div
            className="date-div"
          >
            <div
              className="eventCreation-form"
            >
              <div>
                <label
                  className="date-label"
                  htmlFor="eventDate"
                >
                  Event
                   Starting Date
                </label>
                <input
                  className="form-control mb-2"
                  onChange={[Function]}
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div
              className="eventCreation-form"
            >
              <div>
                <label
                  className="date-label"
                  htmlFor="eventDate"
                >
                  Event
                   Ending Date
                </label>
                <input
                  className="form-control mb-2"
                  onChange={[Function]}
                  type="date"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <select
                className="form-control mb-2"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  Select Venue
                </option>
                <option
                  value="LT1"
                >
                  LT1
                </option>
                <option
                  value="LT2"
                >
                  LT2
                </option>
                <option
                  value="Auditorium"
                >
                  Auditorium
                </option>
                <option
                  value="NCC"
                >
                  NCC
                </option>
                <option
                  value="Other"
                >
                  Other
                </option>
              </select>
            </div>
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="form-label"
                htmlFor="eventDate"
              >
                Event
                 Starting Time
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                placeholder="Event Time"
                type="time"
                value=""
              />
            </div>
          </div>
          <div
            className="eventCreation-form"
          />
          <button
            className="btn btn-primary w-100"
            type="submit"
          >
            Create 
            Event
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
